tests/testthat/test-is_numeric_linter.R

```diff
@@ -0,0 +1,64 @@
+test_that("is_numeric_linter skips allowed usages involving ||", {
+  expect_lint("is.numeric(x) || is.integer(y)", NULL, is_numeric_linter())
+  # x is used, but not identically
+  expect_lint("is.numeric(x) || is.integer(foo(x))", NULL, is_numeric_linter())
+  # not totally crazy, e.g. if input accepts a vector or a list
+  expect_lint("is.numeric(x) || is.integer(x[[1]])", NULL, is_numeric_linter())
+})
+
```

Comment:
Maybe a test case with `inherits()` is also worth including here?

``` r
library(lintr)

lint(
  text = 'inherits(x, c("numeric", "integer"))',
  linters = is_numeric_linter()
)
```

<sup>Created on 2022-10-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
that should lint right? I've never come across this but it looks like is.numeric should apply...

Reply:
I see a TODO about this, so resolving this.